api/rds.proto

```diff
@@ -797,3 +805,22 @@ message HeaderMatcher {
   // * The regex *\d{3}* does not match the value *123.456*
   google.protobuf.BoolValue regex = 3;
 }
+
+// Query parameter matching treats the query string of a request's :path header
+// as an ampersand-separated list of keys and/or key=value elements.
+message QueryParameterMatcher {
+  // Specifies the name of a key that must be present in the requested
+  // *path*'s query string.
+  string name = 1 [(validate.rules).string.min_bytes = 1];
+
+  // Specifies the value of the key. If the value is absent, a request
+  // that contains the key in its query string will match, whether the
+  // key appears with a value (e.g., "?debug=true") or not (e.g., "?debug")
+  string value = 3;
+
+  // Specifies whether the query parameter value is a regular expression.
+  // Defaults to false. The entire query parameter value (i.e., the part to
+  // the right of the equals sign in "key=value") must match the regex.
+  // E.g., the regex "\d+$" will match "123" but not "a123" or "123a".
+  google.protobuf.BoolValue regex = 4;
```

Comment:
I highly recommend we change the design pattern from (string, BoolValue) to oneof(string value, string regex).

The current design has serious usability issues:
- Every read and write has to look at two fields. It is both inconvenient and error prone.
- BoolValue is called dead-end design, because it is impossible to add another variant.
- `"regex": true` in the JSON payload is not obvious to readers. Without remembering the schema, what does regex is true mean to users?
- oneof (value, regex) is both semantically correct and more usable. We can easily add prefix or suffix in the future.
- oneof also has less wire data, because only one field is transferred.